src/jvmMain/kotlin/io/beatmaps/api/maps.kt

```diff
@@ -519,13 +524,16 @@ fun Route.mapDetailRoute() {
                 .joinUploader()
                 .joinCurator()
                 .joinBookmarked(sess?.userId)
+                .joinCollaborators()
                 .select {
                     Beatmap.id.inSubQuery(
                         Beatmap
                             .joinVersions()
+                            .joinCollaborations()
                             .slice(Beatmap.id)
                             .select {
-                                Beatmap.uploader.eq(it.id) and (Beatmap.deletedAt.isNull())
+                                (Beatmap.uploader eq it.id or (Collaboration.collaboratorId eq it.id)) and
```

Comment:
As per my previous comments about the `mapper:` tag in search. I think it would be better to either have a parameter on this endpoint or a new endpoint that enabled bringing in collaborators.